From 08f5312b25049400e522f0ab8ae29c7a28b0ba5e Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:19 -0400
Subject: [scsi] hpsa: make device update copy the raid map also

Message-id: <1410186734-19986-29-git-send-email-thenzl@redhat.com>
Patchwork-id: 92049
O-Subject: [RHEL6.6 e-stor PATCH 28/83] hpsa: make device update copy the raid map also
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Otherwise we could wind up using incorrect raid map data, and
then very bad things would likely happen.

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 9fb0de2d12be813294174127efaa14b8046b1411)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 2bf7ba8..1ab0148 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -909,6 +909,9 @@ static void hpsa_scsi_update_entry(struct ctlr_info *h, int hostno,
  /* Raid offload parameters changed. */
  h->dev[entry]->offload_config = new_entry->offload_config;
  h->dev[entry]->offload_enabled = new_entry->offload_enabled;
+ h->dev[entry]->ioaccel_handle = new_entry->ioaccel_handle;
+ h->dev[entry]->offload_to_mirror = new_entry->offload_to_mirror;
+ h->dev[entry]->raid_map = new_entry->raid_map;
 
  dev_info(&h->pdev->dev, "%s device c%db%dt%dl%d updated.\n",
   scsi_device_type(new_entry->devtype), hostno, new_entry->bus,
-- 
1.7.1